### ISSUE TITLE
Refactoring Main.java into commands

### ISSUE THREAD

**Issue by pimotte** [AsamK/signal-cli#70]
I started to do a little bit of work on making the Main class a bit shorter by extracting functionality into classes per command. The way I did this is somewhat inspired by how I saw/did it in [Syncany](www.github.com/syncany/syncany).

Trial commit can be found here: https://github.com/AsamK/signal-cli/commit/ff4d986c2af5f0adc1b8d26bb964db1c6d03b80a

Initial considerations: 
- Should execute be boolean? Probably not if we throw exceptions to do error handling. Alternative to throwing exceptions would be to have some sort of result-object, including any errors/exceptions.

@AsamK, since this quite a big operation, I would really appreciate any early feedback you might have. (Or if you'd rather not go this way at all, I'd like to hear that too.)

**Comment by AsamK** [comment]
Thanks for taking this on.
I'd go for exceptions and no return values. Currently the signal-cli exit codes indicate the type of error (1=user error, 2=unexpected error, 3=server error). We could use different exceptions to implement these exit codes. 
I'd also suggest, that all fatal errors should throw an exception, to keep the amount of System.err.println code inside the command classes to a minimum.

**Comment by jackschmidt** [comment]
@pimotte I posted some code for a new command to log received messages to JSON. You might check out how the "receive" command is intimately tied up with both Main and Manager. In particular, if I had wanted to keep the standard debug output to System.out, my new command would probably have had to be in Main.

I like splitting the commands out a bit, as it'll make it more clear whether the JSON stuff belonged with Main or Manager or split off like I did. I think AsamK is already writing some JSON stuff, so probably we'll see the right way soon enough. :-)
